modules/data.atmosphere/R/met_temporal_downscale.Gaussian_ensemble.R

```diff
@@ -0,0 +1,398 @@
+    
+    # Temperature For temperature, we create a random normal distribution with the
+    # mean of the source value and the standard deviation of the training value
+    # within the window, and step through that window sequentially so that we are
+    # only using the standard devaition of values recorded at the same time as the
+    # value we are generating.
+    temp <- c("air_temperature", "air_temperature_max", "air_temperature_min")
+    df <- data.frame()
+    for (u in temp) {
+      woo <- vector()
+      a <- as.numeric(train[[u]])
+      sour <- as.numeric(source[[u]])
+      if (all(is.na(sour)) == TRUE) {
+        woo <- rep(NA, reso_len)
+      }
+      if (all(is.na(sour)) == FALSE) {
+        for (x in seq_along(sour)) {
+          four <- vector()
+          for (n in seq_along(1:div)) {
+            lowday <- x * div - wd * sd_step + n
+            highday <- x * div + wd * sd_step + n
+            if (lowday < 0) {
+              lowday <- x + n
+              highday <- x + n + wd * sd_step + n
+            }
+            if (highday > reso_len) {
+              highday <- reso_len
+              lowday <- x * div - wd * sd_step + n
+            }
+            four[n] <- rnorm(1, mean = sour[x], sd = sd(a[seq(from = lowday, 
+                                                              to = highday, by = sd_step)]))
+          }
+          woo <- append(woo, four)
+        }
+      }
+      df[1:length(woo), u] <- woo
+    }
+    
+    # Makes sure that max temps >= avg temp and min temps <= avg temp
+    for (x in seq_along(df$air_temperature)) {
+      if (all(is.na(df$air_temperature_max)) == FALSE) {
+        if (df$air_temperature_max[x] < df$air_temperature[x]) {
+          df$air_temperature_max[x] <- df$air_temperature[x]
+        }
+      }
+      if (all(is.na(df$air_temperature_max)) == TRUE) {
+        df$air_temperature_max <- rep(NA, reso_len)
+      }
+      if (all(is.na(df$air_temperature_min)) == FALSE) {
+        if (df$air_temperature_min[x] > df$air_temperature[x]) {
+          df$air_temperature_min[x] <- df$air_temperature[x]
+        }
+      }
+      if (all(is.na(df$air_temperature_min)) == TRUE) {
+        df$air_temperature_min <- rep(NA, reso_len)
+      }
+    }
+    
+    # Precipitation_flux this takes the daily total of precipitation and uses that as
+    # a total possible amount of precip.  It randomly distributes the values of
+    # precipitation
+    rand_vect_cont <- function(N, M, sd = 1) {
+      vec <- rnorm(N, M/N, sd)
+      vec/sum(vec) * M
+    }
+    precip <- vector()
+    for (x in seq_along(source$precipitation_flux)) {
+      lowday <- (x - wd) * div
+      highday <- (x + wd) * div
+      if (lowday < 0) {
+        lowday <- 0
+      }
+      if (highday > reso_len) {
+        highday <- reso_len
+      }
+      four <- vector()
+      four <- rand_vect_cont(div, source$precipitation_flux[x], sd = sd(train$precipitation_flux[lowday:highday]))
```

Comment:
Ok, so lets separate the issue of which variance to use from how you correct the statistical moments for truncation and focus on the second for now. You are working with a truncated Normal distribution https://en.wikipedia.org/wiki/Truncated_normal_distribution with one sided truncation. That means that the parameters you want to feed into your rnorm are not the sample mean and standard deviations, but rather bias corrected parameters. Look at the wikipedia link and other sources for the exact math on that, but the idea is that to get the correct observed mean and SD you need to feed in different parameters (e.g. in general a lower, and often negative, mu and a higher sigma) so that when you sample from that distribution and then truncate the zeros you end up matching the observed mean and sd